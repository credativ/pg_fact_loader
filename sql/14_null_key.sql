SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';

/***
Based on our config, this should not create an ERROR but should not do anything.
 */
INSERT INTO test.orders (order_id, customer_id, order_date, total)
VALUES (5, NULL, '2018-04-10', 100.00);

SELECT COUNT(1) FROM test_audit_raw.orders_audit;

/****
We limit everything to this 1 table because the above grossly violates our schema and will create
errors on other tables.  We just want to verify that this actually runs without error when processed.
 */
UPDATE fact_loader.fact_tables SET force_worker_priority = TRUE WHERE fact_table_relid = 'test_fact.orders_fact'::REGCLASS;

SELECT test.tick();
SELECT fact_loader.worker() FROM generate_series(1,6);

SELECT order_id, customer_id, order_date, total, is_reorder
FROM test_fact.orders_fact
ORDER BY order_id;

SELECT fact_loader.purge_queues('0 seconds'::INTERVAL);

SELECT COUNT(1) FROM test_audit_raw.orders_audit;

TRUNCATE test_audit_raw.orders_audit;

UPDATE fact_loader.fact_tables SET force_worker_priority = FALSE WHERE fact_table_relid = 'test_fact.orders_fact'::REGCLASS;