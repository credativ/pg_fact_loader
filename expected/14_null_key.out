SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';
/***
Based on our config, this should not create an ERROR but should not do anything.
 */
INSERT INTO test.orders (order_id, customer_id, order_date, total)
VALUES (5, NULL, '2018-04-10', 100.00);
SELECT COUNT(1) FROM test_audit_raw.orders_audit;
 count 
-------
     1
(1 row)

/****
We limit everything to this 1 table because the above grossly violates our schema and will create
errors on other tables.  We just want to verify that this actually runs without error when processed.
 */
UPDATE fact_loader.fact_tables SET force_worker_priority = TRUE WHERE fact_table_relid = 'test_fact.orders_fact'::REGCLASS;
SELECT test.tick();
 tick 
------
 
(1 row)

SELECT fact_loader.worker() FROM generate_series(1,6);
 worker 
--------
 t
 t
 t
 t
 t
 t
(6 rows)

SELECT order_id, customer_id, order_date, total, is_reorder
FROM test_fact.orders_fact
ORDER BY order_id;
 order_id | customer_id | order_date |  total  | is_reorder 
----------+-------------+------------+---------+------------
        1 |           1 | 04-10-2018 |  100.00 | f
        3 |           5 | 04-12-2018 | 2000.00 | f
        4 |           1 | 04-13-2018 |  100.00 | f
(3 rows)

SELECT fact_loader.purge_queues('0 seconds'::INTERVAL);
 purge_queues 
--------------
 
(1 row)

SELECT COUNT(1) FROM test_audit_raw.orders_audit;
 count 
-------
     1
(1 row)

TRUNCATE test_audit_raw.orders_audit;
UPDATE fact_loader.fact_tables SET force_worker_priority = FALSE WHERE fact_table_relid = 'test_fact.orders_fact'::REGCLASS;
